src/bases.jl

```diff
@@ -1,268 +1,3 @@
-"""
```

Comment:
This whole file was moved to `QuantumInterface`. Basically everything that has to do with `Basis` is now there. This is the only code that contains logic, not just naming conventions, that was moved.

I also moved the most common bases there, like FockBasis and PauliBasis. I did not move all of them, as some had extra machinery attached to them that I did not want to immediately try to decouple.